File: bisectionMethod/bisection.js
// Define the quadratic equation: x^3 - 4x - 9
function equation(x) {
    return x * x * x - 4 * x - 9;
}

// Bisection method function
function bisection(func, a, b, tolerance, maxIterations) {
    for (let i = 1; i <= maxIterations; i++) {
        let root = (a + b) / 2.0;
        let fx = func(root);

        console.log(i + "\t\t" + root + "\t" + fx);

        if (Math.abs(fx) < tolerance) {
            console.log("\nRoot found: " + root);
            return root; // Return the root when found within tolerance
        }

        if (fx * func(a) < 0) {
            b = root; // Update the interval
        } else {
            a = root; // Update the interval
        }
    }

    console.log("\nRoot not found within " + maxIterations + " iterations.");
    return NaN; // Return NaN if root not found
}

// Main function
function main() {
    const a = 1.0; // Left endpoint of the interval
    const b = 3.0; // Right endpoint of the interval
    const tolerance = 1e-6; // Tolerance for stopping criterion
    const maxIterations = 100; // Maximum number of iterations

    console.log("Iteration\tRoot\tf(x)");

    bisection(equation, a, b, tolerance, maxIterations);
}

main();

File: bisectionMethod/bisection.c
#include <stdio.h>
#include <math.h>

// Define the quadratic equation: x^3 - 4x - 9
float equation(float x)
{
    return (x * x * x) - (9* x) +1;
}

// Bisection method function
float bisection(float func(float), float a, float b, float tolerance, int maxIterations)
{
    for (int i = 1; i <= maxIterations; i++)
    {
        float root = (a + b) / 2.0;
        float fx = func(root);

        printf("%d\t\t%.6f\t%.6f\n", i, root, fx);

        if (fabs(fx) < tolerance)
        {
            printf("\nRoot found: %.6f\n", root);
            return root; // Return the root when found within tolerance
        }

        if (fx * func(a) < 0)
        {
            b = root; // Update the interval
        }
        else
        {
            a = root; // Update the interval
        }
    }

    printf("\nRoot not found within %d iterations.\n", maxIterations);
    return NAN; // Return NaN if root not found
}

int main()
{
    float a = 1.0;           // Left endpoint of the interval
    float b = 3.0;           // Right endpoint of the interval
    float tolerance = 1e-6;  // Tolerance for stopping criterion
    int maxIterations = 100; // Maximum number of iterations

    printf("Iteration\tRoot\tf(x)\n");

    bisection(equation, a, b, tolerance, maxIterations);

    return 0;
}
